status/adapter.go

```diff
@@ -0,0 +1,49 @@
+package status
+
+import (
+	"encoding/json"
+
+	"github.com/awslabs/operatorpkg/object"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// UnstructuredAdapter is an adapter for the status.Object interface. unstructuredAdapter
+// makes the assumption that status conditions are found on status.conditions path.
+type UnstructuredAdapter struct {
+	*unstructured.Unstructured
+}
+
+func NewUnstructuredAdapter(obj client.Object) (*UnstructuredAdapter, error) {
+	u := &unstructured.Unstructured{Object: lo.Must(runtime.DefaultUnstructuredConverter.ToUnstructured(obj))}
```

Comment:
Checking the status conditions and validating that they exist on the object doesn't work as far as I can tell since conditions are not always present, meaning that we can really only validate it at runtime and assume that the developer correctly chooses objects that have status conditions on "status.conditions"